File: impl/cpp/Item.cpp
using namespace std;

#include "Item.h"

template <typename T>
Item<T>::Item(T k) {
    this->key = k;
    this->next = this;
}

template <typename T>
T Item<T>::getKey() {
    return key;
}

template <typename T>
void Item<T>::setKey(T k) {
    key = k;
}

template <typename T>
Item<T> Item<T>::getNext() {
    return next;
}

template <typename T>
void Item<T>::setNext(Item<T> n) {
    next = n;
}

File: impl/java/src/softheap/Item.java
package softheap;

public class Item<T extends Comparable<T>> {
    private T key;
    private Item<T> next;

    public Item(T key) {
        this.key = key;
        this.next = this;
    }

    public T getKey() {
        return key;
    }

    public void setKey(T key) {
        this.key = key;
    }

    public Item<T> getNext() {
        return next;
    }

    public void setNext(Item<T> next) {
        this.next = next;
    }
}
